lib/travis/build/script/dart.rb

```diff
@@ -230,12 +230,17 @@ def dartanalyzer
           def dartfmt
             args = task[:dartfmt]
 
-            if args.is_a?(String)
-              sh.echo "dartfmt arguments aren't supported.", ansi: :red
-            end
-
-            sh.if package_direct_dependency?('dart_style'), raw: true do
-              sh.raw 'function dartfmt() { pub run dart_style:format "$@"; }'
+            # If specified `-dartfmt: sdk` and there is a dependency on `dart_style` we
+            # will use the SDK version of dart_style to run formatting checks instead of
+            # the custom pinned version.
+            if args != 'sdk'
+              sh.if package_direct_dependency?('dart_style'), raw: true do
+                sh.echo 'Using the provided dart_style package to run format instead of the SDK.'
+                sh.echo "You may specify '- dartfmt: sdk' in order to use the SDK version instead"
+                sh.raw 'function dartfmt() { pub run dart_style:format "$@"; }'
+              end
+            elsif args.is_a?(String)
+              sh.echo "dartfmt only supports 'sdk' as an optional argument value.", ansi: :red
             end
```

Comment:
I don't think the conditions are ok.
If I use `dartfmt: sdk` the first `if` will be `false` and the `elsif` true then the message _dartfmt only supports 'sdk' as an optional argument value_ will be displayed.
If I use `dartfmt: something` the first `if` will be `true`.
If I use `dartfmt` it seems to be ok.